README.md

```diff
@@ -1,3 +1,280 @@
+orgs with Dev Hub or in sandbox orgs.  Follow the steps on 
+[this page](https://developer.salesforce.com/docs/platform/functions/guide/configure_your_org.html) to ensure everything 
+is setup correctly first.
+
+Once your Org is configured you can login and set it as the default Dev Hub with the following command:
+
+```sh
+sf login org --alias PythonOrg --set-default-dev-hub --set-default
+```
+
+This will make **PythonOrg** the default Dev Hub for subsequent commands.
+
+Then create a scratch org:
+
+```sh
+sfdx force:org:create \
+  --definitionfile config/project-scratch-def.json \
+  --setalias PythonScratch \
+  --setdefaultusername
+```
+
+This will make it so that, when you run a Function, it will connect to and use the **PythonScratch** org.
+
+### Connect Your Compute Environment
+
+Login to Salesforce Functions with the same credentials you used to connect your Dev Hub org.
+
+```sh
+sf login functions
+```
+
+Then you will be able to create the compute environment and associate it with the **PythonScratch** org we created while
+connecting your org.
+
+```sh
+sf env create compute \
+  --connected-org PythonScratch \
+  --alias PythonCompute
+```
+
+This will make it so that, when you deploy a Function, it will be deployed to the *PythonCompute* environment linked to 
+your scratch org.
+
+## Create and Run a Python Function Locally
+
+### Generate the Python Function
+
+From the SFDX project root, run:
+
+```sh
+sf generate function \
+  --language python \
+  --function-name hellopython
+```
+
+The remaining commands will be executed within the newly generated project folder change to that folder by running
+
+```sh
+cd functions/hellopython
+```
+
+### Create the Python [Virtual Environment](https://docs.python.org/3/library/venv.html#creating-virtual-environments) & Install Dependencies
+
+```sh
+python3 -m venv venv
+source venv/bin/activate
+pip3 install -r requirements.txt
+```
+
+This will ensure your function has all dependencies it requires installed before you run it.  If you forgot to do this 
+before starting the Function locally you’ll receive an error reminding you to perform this step.
+
+### Run the Python Function Locally
+
+```sh
+sf run function start
+```
+
+This will start the function running locally on http://localhost:8080.  Messages logged by the running function will appear here.
+
+### Invoke the Running Python Function
+
+After starting the function:
+
+* open a new command line terminal 
+* navigate to the `hellopython` directory 
+* invoke the function by sending it a payload
+
+```sh
+sf run function --function-url http://localhost:8080 --payload '{}'
+```
+
+## Deploy the Python Function
+
+The following commands need to be executed from the **root of your SFDX project** so change into that directory now.
+
+### Commit your changes to Git
+
+All code changes made to a function will need to be staged and committed before you can deploy.
+
+```sh
+git add . 
```

Comment:
Ah so it's just occurred to me that the steps as written in this doc will have them git pushing the `venv` when they shouldn't. Whilst there is a GUS work item filed for coming up with something better longer term (eg switching to Poetry, or adding a `.gitignore` to the template), I think for this doc we need to either:
1. Have them create the venv outside of the project (eg `~/venvs/hellopython`)
2. or else have them `echo "venv/" > .gitignore` prior to running `git add`

I think perhaps (2) is best, since creating the venv inside the project at least means VSCode will auto-detect it whenever they open a console etc.